### PR TITLE
fix(rds): fail with a descriptive error if Cluster's instance count is a deploy-time value

### LINKED ISSUE #13558: (RDS): deploy-time value can't be used for nr of instances in Cluster
I'm attempting to store some configuration variables in parameter store for my RDS cluster but am running into [this validation error](https://github.com/aws/aws-cdk/blob/6c5b1f42fb73a132d47945b529bab73557f2b9d8/packages/%40aws-cdk/aws-docdb/lib/cluster.ts#L370) about instance size needing to be at least 1. It's comparing the token's value, which is a big negative number. 

### Reproduction Steps

```typescript
import * as rds from "@aws-cdk/aws-rds";
import * as ec2 from "@aws-cdk/aws-ec2";
import * as ssm from "@aws-cdk/aws-ssm";


class DbStack extends Stack {
    vpc: ec2.Vpc;

    constructor(scope: Construct, id: string, props?: StackProps) {
      this.vpc = new ec2.Vpc(this, "GrowMeVpc", {
        maxAzs: 3,
        enableDnsSupport: true,
        enableDnsHostnames: true,
      });

      const RDS_INSTANCES = Token.asNumber(getSsmStringParam("RDS_INSTANCES"));

        new rds.DatabaseCluster(this, "DbCluster", {
          engine: rds.DatabaseClusterEngine.auroraPostgres({
            version: rds.AuroraPostgresEngineVersion.of("12.4", "12", {
              s3Import: true,
              s3Export: true,
            }),
          }),
          defaultDatabaseName: "db",
          instances: RDS_INSTANCES,
          instanceProps: {
            instanceType: ec2.InstanceType.of(
             "t3",
             "medium"
            ),
            vpcSubnets: {
              subnetType: ec2.SubnetType.PRIVATE,
            },
            vpc: this.vpc,
          },
      });
  }
    
```

### What did you expect to happen?

I would expect the validation to run after the token has resolved only

### What actually happened?

Validation ran on the token's value itself

### Environment

  - **CDK CLI Version  :** 1.92.0
  - **Framework Version:** 1.92.0
  - **Node.js Version:** 14.16.0
  - **OS               :** Max OS
  - **Language (Version):** TypeScript 4.2.2

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by BLasan**
Added a condition to check whether the `instanceCount` is a token or not. If it's not a token then an exception will be thrown.

Fixes #13558 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by BLasan** [comment]
@skinny85 Please review

File: packages/@aws-cdk/aws-rds/lib/cluster.ts [hunk 1]
```diff
@@ -651,6 +651,9 @@ interface InstanceConfig {
  */
 function createInstances(cluster: DatabaseClusterNew, props: DatabaseClusterBaseProps, subnetGroup: ISubnetGroup): InstanceConfig {
   const instanceCount = props.instances != null ? props.instances : 2;
+  if (Token.isUnresolved(instanceCount)) {
+    throw new Error('Instance count is not a token');
```
Comment (skinny85) [inline packages/@aws-cdk/aws-rds/lib/cluster.ts:655]: This error message needs to be better here. Maybe something like:

```suggestion
    throw new Error('The number of instances an RDS Cluster consists of cannot be provided as a deploy-time only value!');
```

**Review by skinny85 — CHANGES_REQUESTED** [review]
Thanks for the contribution @BLasan!

You're missing an unit test for the fix though 🙂.

**Comment by BLasan** [comment]
@skinny85 Made the requested changes. Please review

**Review by skinny85 — APPROVED** [review]
Looks good @BLasan , thanks for the contribution!

**Comment by BLasan** [comment]
@skinny85 Build fails

**Review by skinny85 — APPROVED** [review]
This time should do it 😉

**Comment by BLasan** [comment]
> This time should do it wink

thanks :)

**Pull request merged**